src/utils/teams.spec.ts

```diff
@@ -0,0 +1,196 @@
+import sinon from 'sinon';
+import request from '../request.js';
+import { sinonUtil } from './sinonUtil.js';
+import { formatting } from './formatting.js';
+import { teams } from './teams.js';
+import assert from 'assert';
+import { cli } from '../cli/cli.js';
+import { settingsNames } from '../settingsNames.js';
+
+const teamName = 'HR Team';
+const teamId = '0b0b204f-7ca0-4c7f-baf2-53caa381828b';
+const teamResponse = {
+  id: teamId
+};
+const channelName = 'General';
+const channelId = '19:7a3a82caa8f8436889fbb017acdb11b6@thread.tacv2';
+const channelResponse = { id: channelId };
+
+describe('utils/teams', () => {
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      cli.getSettingWithDefaultValue,
+      teams.getTeamIdByDisplayName,
+      teams.verifyChannelExistsById,
+      teams.getChannelIdByDisplayName,
+      teams.getTeamId,
+      teams.verifyTeamExistsById
+    ]);
+  });
+
+  it('gets team id by displayName', async () => {
+    sinon.stub(teams, 'getTeamIdByDisplayName').resolves(teamId);
+
+    const actual = await teams.getTeamId(undefined, teamName);
+    assert.strictEqual(actual, teamId);
+  });
+
+  it('returns team id and verifies that team exists', async () => {
+    sinon.stub(teams, 'verifyTeamExistsById').resolves(teamId);
+
+    const actual = await teams.getTeamId(teamId, undefined);
+    assert.strictEqual(actual, teamId);
+  });
+
+  it('verifies that team exists by id', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/teams/${teamId}?$select=id`) {
+        return { id: teamId };
+      }
+
+      throw 'Invalid Request';
+    });
+
+    const actual = await teams.verifyTeamExistsById(teamId);
+    assert.strictEqual(actual, teamId);
+  });
+
+  it('correctly get team id by displayName', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/teams?$filter=displayName eq '${formatting.encodeQueryParameter(teamName)}'&$select=id`) {
+        return { value: [teamResponse] };
+      }
+
+      throw 'Invalid Request';
+    });
+
+    const actual = await teams.getTeamIdByDisplayName(teamName);
+    assert.strictEqual(actual, teamId);
+  });
+
+  it('throws error if no teams are found', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/teams?$filter=displayName eq '${formatting.encodeQueryParameter(teamName)}'&$select=id`) {
+        return { value: [] };
+      }
+
+      throw 'Invalid Request';
+    });
+
+    await assert.rejects(teams.getTeamIdByDisplayName(teamName), Error(`The specified team '${teamName}' does not exist.`));
+  });
+
+  it('throws error message when multiple teams were found using getTeamIdByDisplayName', async () => {
+    sinon.stub(cli, 'getSettingWithDefaultValue').callsFake((settingName, defaultValue) => {
```

Comment:
Let's set this stub in a `before` function and let's refactor it a bit to an inline if: `.callsFake((settingName, default) => settingName === settingsNames.prompt ? false : default)`